src/query/functions/tests/it/scalars/map.rs

```diff
@@ -144,3 +146,37 @@ fn test_map_values(file: &mut impl Write) {
         &columns,
     );
 }
+
+fn test_map_cat_ops_basic(file: &mut impl Write) {
+    let columns = [
+        ("a_col", StringType::from_data(vec!["a_k1", "a_k2", "a_k3"])),
+        ("b_col", StringType::from_data(vec!["b_k1", "b_k2", "b_k3"])),
+        ("c_col", StringType::from_data(vec!["c_k1", "c_k2", "c_k3"])),
+        ("d_col", StringType::from_data(vec!["aaa1", "aaa2", "aaa3"])),
+        ("e_col", StringType::from_data(vec!["bbb1", "bbb2", "bbb3"])),
+        ("f_col", StringType::from_data(vec!["ccc1", "ccc2", "ccc3"])),
+    ];
+
+    run_ast(
+        file,
+        "map_cat(map([a_col, b_col], [d_col, e_col]), map([c_col], [f_col]))",
+        &columns,
+    );
+}
+
+fn test_map_cat_impl2_ops_basic(file: &mut impl Write) {
+    let columns = [
+        ("a_col", StringType::from_data(vec!["a_k1", "a_k2", "a_k3"])),
+        ("b_col", StringType::from_data(vec!["b_k1", "b_k2", "b_k3"])),
+        ("c_col", StringType::from_data(vec!["c_k1", "c_k2", "c_k3"])),
```

Comment:
add a case that the second map have a same key with the first map, such as `a_k3`, so that we can test the output map have the value from the second map.

Reply:
add please also add some sql logic tests this case, for example
```sql
MySQL [default]> select map_cat({'k1':'v1','k2':'v2'}, {'k1':'abc'});
+-----------------------------------------------+
| map_cat2({'k1':'v1','k2':'v2'}, {'k1':'abc'}) |
+-----------------------------------------------+
| {'k1':'abc','k2':'v2'}                        |
+-----------------------------------------------+
1 row in set (0.026 sec)
```

Reply:
Sure, got it.